src/audio/base_fw.c

```diff
@@ -252,6 +263,15 @@ static uint32_t basefw_set_system_time(uint32_t param_id,
 	global_system_time_info.dsp_time.val_l = (uint32_t)(current_dsp_time);
 	global_system_time_info.dsp_time.val_u = (uint32_t)(current_dsp_time >> 32);
 
+	/* use default timestamp if 64bit is not enabled since 64bit is necessary for host time */
+	if (!IS_ENABLED(CONFIG_LOG_TIMESTAMP_64BIT))
+		return IPC4_SUCCESS;
+
+	host_time = k_ns_to_cyc_ceil64(*(uint64_t *)data + TIME_SETTING_LATENCY_NS);
```

Comment:
This is very strange: 10 lines earlier `data` is cast to `struct ipc4_system_time`. But here the same `data` is cast to `uint64_t`?

Reply:
Sorry I didn't make myself clear. My main question is: why is `data` going to two different places? Why isn't one place enough?

Reply:
to simplify the code,  I don't want to pack two 32bit number to 64bit one, so reuse data